library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -442,9 +443,16 @@ private Slice readOrSkip(boolean skip) throws IOException
           inputStream = getInputStream(currentPointer);
         }
 
-        if (inputStream != null && currentPointer.offset <
-            fileSystemWAL.fileContext.getFileStatus(currentOpenPath).getLen()) {
-          int len = inputStream.readInt();
+        int ch1;
+
+        if (inputStream != null && (ch1 = inputStream.read()) != -1) {
```

Comment:
There is a logic to read the next segment, which will be skipped if the exception is thrown at the EOF. So EOFException is required only when the write to WAL was incomplete, which also indicates no next Segment.

Another aspect I was looking was, EOF should not be indicated by an Exception (Effective Java- Item 57)